src/Components/README.md

```diff
@@ -1,17 +1,53 @@
 # Blazor
 
-**Build modern, interactive web-based UIs with C# and Razor.**
+Blazor is a framework for building modern, interactive web-based UIs with C# and Razor. For more information on using Blazor to build apps, check out [the official docs](https://blazor.net).
 
-This folder contains the underlying *components* programming model that powers both server-side and client-side [Blazor](#blazor) applications.
+## Description
 
-Blazor is a component based web UI framework. Blazor apps can run client-side in browser on WebAssembly or server-side as part of an ASP.NET Core app. Blazor is a full single-page application (SPA) framework inspired by the latest JavaScript SPA frameworks, featuring support for offline/PWA applications, app size trimming, and browser-based debugging.
+This folder contains the component model shared between the WebAssembly and Server hosting models for Blazor.
 
-Blazor uses only the latest web standards. No plugins or transpilation needed. It runs in the browser on a real .NET runtime implemented in [WebAssembly](http://webassembly.org) that executes normal .NET assemblies.
+The following contains a description of each sub-directory in the `Components` directory.
 
-[![Gitter](https://badges.gitter.im/aspnet/blazor.svg)](https://gitter.im/aspnet/blazor?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
+- `Analyzers`: Contains a collection of Rosyln analyzers for Blazor components
+- `Authorization`: Contains source files associated with auth-related components and services in Blazor
+- `Components`: Contains the implementation for the Blazor component model
+- `Forms`: Contains source files for Form components in Blazor
+- `Ignitor`: Contains source files for inspecting Blazor apps
+- `Samples`: Contains a collection of sample apps in Blazor
+- `Server`: Contains the implementation for Blazor Server-specific components
+- `Shared`: Contains a collection of shared constants and helper methods/classes
+- `Web`: Contains source files for handling DOM events, forms, and other components
+- `Web.JS`: Contains the source files for Blazor's client-side JavaScript
+- `WebAssembly`: Contains the implementation for WebAssembly-specific components
```

Comment:
Should we also document individual folders inside this directory? There's half a dozen of these.

Reply:
Yep. Added some content here.